### PR TITLE
feat: add handling for isArray in form field model

### PR DESCRIPTION
**Pull request by awinberg-aws**
## Problem
Studio model now has an isArray property for form fields which isn't being handled when generating form definitions

## Solution
Update type and helper methods to include support for isArray.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: packages/codegen-ui-react/lib/utils/forms/array-field-component.ts [hunk 1]
```diff
@@ -1031,7 +1030,7 @@ export const renderArrayFieldComponent = (fieldName: string, fieldLabel: string,
         ),
         factory.createJsxAttribute(
           factory.createIdentifier('inputFieldRef'),
-          factory.createJsxExpression(undefined, factory.createIdentifier(`${lowerCaseFirst(fieldName)}Ref`)),
+          factory.createJsxExpression(undefined, factory.createIdentifier(`${fieldName}Ref`)),
```
Comment (awinberg-aws) [inline packages/codegen-ui-react/lib/utils/forms/array-field-component.ts:1033]: Resolves undefined error. The ref being referenced here is not declared using lowerCaseFirst.

https://github.com/aws-amplify/amplify-codegen-ui/blob/main/packages/codegen-ui-react/lib/forms/react-form-renderer.ts#L435

File: packages/codegen-ui/lib/generate-form-definition/helpers/form-field.ts [hunk 2]
```diff
@@ -339,6 +339,7 @@ export function getFormDefinitionInputElement(
 
   formDefinitionElement.validations = mergedValidations;
   formDefinitionElement.dataType = config?.dataType || baseConfig?.dataType;
+  formDefinitionElement.isArray = config.inputType?.isArray || baseConfig?.inputType?.isArray;
```
Comment (cwoolum) [inline packages/codegen-ui/lib/generate-form-definition/helpers/form-field.ts:343]: I think we want the DataStore `isArray` config to take priority. If a user has a DataStore field that is just a scalar, they shouldn't be able to override that with an array but that might be a @renebrandel question.

Comment (cwoolum) [reply packages/codegen-ui/lib/generate-form-definition/helpers/form-field.ts:343]: Similarly for the reverse, if the field is a scalar and the user configures it as an array, does the use have to map it correctly in code?

**Review by cwoolum — COMMENTED** [review]
Have some questions for you and Rene

File: packages/codegen-ui/lib/__tests__/generate-form-definition/generate-form-definition.test.ts [hunk 3]
```diff
@@ -398,6 +413,7 @@ it('should add read-only fields if it has overrides', () => {
     name: {
       componentType: 'TextField',
       dataType: 'String',
+      isArray: false,
```
Comment (hein-j) [inline packages/codegen-ui/lib/__tests__/generate-form-definition/generate-form-definition.test.ts:416]: I think you have to let the metadata get its `isArray` field from the form def element now, or it won't get picked up during codegen. The change is in `form-component-metadata.ts`

File: packages/codegen-ui/lib/generate-form-definition/helpers/form-field.ts (hunk 2 above)
Comment (awinberg-aws) [reply packages/codegen-ui/lib/generate-form-definition/helpers/form-field.ts:343]: new version prioritizes value in baseConfig if present

File: packages/codegen-ui/lib/__tests__/generate-form-definition/generate-form-definition.test.ts (hunk 3 above)
Comment (awinberg-aws) [reply packages/codegen-ui/lib/__tests__/generate-form-definition/generate-form-definition.test.ts:461]: discussed offline. change was not needed based on previous implementation, but a new version was pushed which no longer relies on referencing dataSchema.

**Review by cwoolum — APPROVED** [review]
LGTM

**Review by hein-j — APPROVED** [review]

**Pull request merged**